soc/arm/microchip_mec/common/soc_dma.c

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright (c) 2021 Microchip Technology Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <sys/sys_io.h>
+#include <sys/__assert.h>
+#include <soc.h>
+
+/* #define SOC_DMA_DEBUG_ENABLE 1 */
+
+int soc_dma_init(void)
+{
+	mec_pcr_periph_slp_ctrl(PCR_DMA, 0);
+
+	struct dma_regs *regs = (struct dma_regs *)(DMA_BASE);
+
+	regs->ACTRST = MCHP_DMAM_CTRL_SOFT_RESET | MCHP_DMAM_CTRL_ENABLE;
+
+	uint32_t cnt = 256U;
```

Comment:
How long is this timeout, really? There is no check so won't it depend on the speed of the CPU?